### PR TITLE
feat: mutable barrier interval

### PR DESCRIPTION
**Pull request by Gun9niR**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

`barrier_interval_ms` is used in several places

- Meta node
  - `GlobalBarrierManager`: schedule barrier
- Compute
  - `GlobalMemoryManager`: check eviction every barrier_interval.
  - Executors: pause source
    - `SourceExecutor`
    - `FsSourceExecutor`

I have manually tested setting barrier interval to values ranging from 500ms to 30s, which seems to be working fine.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/compute/src/memory_management/memory_manager.rs [hunk 1]
```diff
@@ -190,4 +204,10 @@ impl GlobalMemoryManager {
             self.set_watermark_time_ms(watermark_time_ms);
         }
     }
+
+    fn get_eviction_check_interval(params: &SystemParamsReaderRef) -> u64 {
+        // Arbitrarily set a minimal barrier interval in case it is too small,
+        // especially when it's 0.
+        std::cmp::max(params.load().barrier_interval_ms() as u64, 10)
+    }
```
Comment (xx01cyx) [inline src/compute/src/memory_management/memory_manager.rs:212]: May we print a warning somewhere if the user sets a too-small `barrier_interval_ms`?

**Review by xx01cyx — COMMENTED** [review]
Generally LGTM.

File: src/compute/src/memory_management/memory_manager.rs (hunk 1 above)
Comment (Gun9niR) [reply src/compute/src/memory_management/memory_manager.rs:212]: We can check the value range on `ALTER SYSTEM SET`, the question is how small is *too-small* 🤣.

File: src/compute/src/memory_management/memory_manager.rs [hunk 2]
```diff
@@ -109,12 +110,25 @@ impl GlobalMemoryManager {
...
+                Ok(_) = params_rx.changed() => {
+                    let new_barrier_interval_ms = Self::get_eviction_check_interval(&params_rx.borrow());
+                    if new_barrier_interval_ms != barrier_interval_ms
+                     {
+                        tick_interval = tokio::time::interval(Duration::from_millis(new_barrier_interval_ms));
+                        barrier_interval_ms = new_barrier_interval_ms;
+                    }
+                }
+                _ = tick_interval.tick() => {},
+            }
```
Comment (liurenjie1024) [inline src/compute/src/memory_management/memory_manager.rs:131]: Please don't change this. Make barrier interval too large may lead to oom.

Comment (Gun9niR) [reply src/compute/src/memory_management/memory_manager.rs:131]: Shall we have another config for this interval?

Comment (xx01cyx) [reply src/compute/src/memory_management/memory_manager.rs:131]: The eviction of the LRU cache (i.e. streaming memory control) actually still depends on barriers, so simply having another config for this checking interval may not help. If we do want to make barrier interval configurable, we should make sure our users know what they are doing (including the side effects). cc. @fuyufjh

**Comment by Gun9niR** [comment]
As discussed offline, barrier interval should not be changed 🤣

**Pull request closed without merging**